test/test_collection.py

```diff
@@ -32,6 +35,27 @@ def failed(self, event):
         pass
 
 class TestExplainableCollection(unittest.TestCase):
+    def _compare_command_dicts(self, ours, theirs):
+        try:
+            for key in ours.keys():
+                if isinstance(ours[key], dict) or isinstance(ours[key], SON):
+                    self._compare_command_dicts(ours[key], theirs[key])
+                elif type(ours[key]) == list:
+                    for i, j in zip(ours[key], theirs[key]):
+                        if isinstance(i, dict) or isinstance(i,
+                                                                     SON):
+                            self._compare_command_dicts(i, j)
+                        else:
+                            assert i == j
+                else:
+                    assert ours[key] == theirs.get(key, None)
+        except:
+            print("ours:", sorted(ours.items()))
+            print("theirs:", sorted([i for i in theirs.items() if ours.get(i[
+                                                                               0], None)
+            != None]))
```

Comment:
Is this debug code? Should the except block re-raise the exception? What if it raised an AssertionError with the "ours/theirs" string instead of printing?

Reply:
It is debug code, I removed it